### PR TITLE
BUG: None / Timedelta incorrectly returning NaT

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

xref #31869, arguable whether or not this closes that.

### CODE REVIEW

**Comment by WillAyd** [comment]
Not against the design just curious as to what your thought is as to why this should raise instead of returning NaN? Perhaps because np.nan / pd.NaT is ambiguous?

**Comment by jbrockmendel** [comment]
> why this should raise instead of returning NaN? Perhaps because np.nan / pd.NaT is ambiguous?

Think of it as if we had non-NA scalars of the same types/dtypes.  e.g. `2.0 / Timedelta(days=1)` is obviously gibberish.  So treat `np.nan / Timedelta(days=1)` the same

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 1]
```diff
@@ -412,6 +412,46 @@ def test_td_rdiv_timedeltalike_scalar(self):
 
         assert np.timedelta64(60, "h") / td == 0.25
 
+    def test_td_rdiv_na_scalar(self):
+        # GH#31869 None gets cast to NaT
+        td = Timedelta(10, unit="d")
+
+        result = NaT / td
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: maybe as a follown split out the these non-error casees and use nulls_fixture (as you then can test float('nan')

**Comment by jreback** [comment]
thanks

**Pull request merged**